src/boot/bootctl-status.c

```diff
@@ -163,24 +241,53 @@ static int status_variables(void) {
         else if (n_order < 0)
                 return log_error_errno(n_order, "Failed to read EFI boot order: %m");
 
-        /* print entries in BootOrder first */
-        for (int i = 0; i < n_order; i++)
-                (void) print_efi_option(order[i], &n_printed, /* in_order= */ true);
+        /* print entries in BootOrder first or add to JsonVariant for Json output */
+        if (arg_json_format_flags != JSON_FORMAT_OFF) {
+                for (int i = 0; i < n_order; i++)
+                        (void) print_efi_option(order[i], &n_printed, /* in_order= */ true, &array);
 
-        /* print remaining entries */
-        for (int i = 0; i < n_options; i++) {
-                for (int j = 0; j < n_order; j++)
-                        if (options[i] == order[j])
-                                goto next_option;
+                /* print remaining entries */
+                for (int i = 0; i < n_options; i++) {
+                        for (int j = 0; j < n_order; j++)
+                                if (options[i] == order[j])
+                                        goto next_option_json;
 
-                (void) print_efi_option(options[i], &n_printed, /* in_order= */ false);
+                        (void) print_efi_option(options[i], &n_printed, /* in_order= */ false, &array);
+                next_option_json:
+                        continue;
+                }
 
-        next_option:
-                continue;
-        }
+                r = json_variant_merge_objectb(&v, JSON_BUILD_OBJECT(
+                        JSON_BUILD_PAIR("BootLoadersEFIVariables", JSON_BUILD_VARIANT(array))));
+                if (r < 0)
+                        return log_oom();
+                *ret_variant = TAKE_PTR(v);
+        } else {
+                for (int i = 0; i < n_order; i++)
+                        (void) print_efi_option(order[i], &n_printed, /* in_order= */ true, NULL);
 
-        if (n_printed == 0)
-                printf("No boot loaders listed in EFI Variables.\n\n");
+                /* print remaining entries */
+                for (int i = 0; i < n_options; i++) {
+                        for (int j = 0; j < n_order; j++)
+                                if (options[i] == order[j])
+                                        goto next_option;
+
+                        (void) print_efi_option(options[i], &n_printed, /* in_order= */ false, NULL);
+                next_option:
+                        continue;
+                }
+        }
+        if (n_printed == 0) {
+                if (arg_json_format_flags != JSON_FORMAT_OFF) {
+                        r = json_variant_merge_objectb(&v, JSON_BUILD_OBJECT(
+                                JSON_BUILD_PAIR("BootLoadersEFIVariables", JSON_BUILD_CONST_STRING("No boot loaders listed in EFI Variables."))));
```

Comment:
Same problem already mentioned earlier: this is writing sentences as json payloads. As already mentioned, json payloads needs to contain data, not sentences, so please go through all the json data you are adding, and ensure it all contains data, rather than log-like strings.